velox/docs/functions/spark/math.rst

```diff
@@ -108,11 +108,30 @@ Mathematical Functions
 
 .. spark:function:: rand() -> double
 
-    Returns a random value with independent and identically distributed uniformly distributed values in [0, 1). ::
+    Returns a random value with identically distributed uniformly distributed values in [0, 1). ::
 
         SELECT rand(); -- 0.9629742951434543
-        SELECT rand(0); -- 0.7604953758285915
-        SELECT rand(null); -- 0.7604953758285915
+
+.. spark:function:: rand(seed, partitionIndex) -> double
+
+    Returns a random value with identically distributed uniformly distributed values in [0, 1) using
+    a combined seed. ``seed``, specified by end user, plus ``partitionIndex``, assigned by framework,
+    will serve as the actual seed for generator. With a same ``seed``, each thread should receive a
+    different ``partitionIndex`` to produce a different sequence of random numbers for corresponding
+    partition.
+    It requires each partition's ``partitionIndex`` is unique & fixed at each run.
+    NULL ``seed`` is identical to zero ``seed``. ``partitionIndex`` cannot be NULL. ::
```

Comment:
Let's also clarify that 'seed' must be constant for all rows (e.g. cannot be a column value).